### ISSUE TITLE
Evaluation function and the number of fine-tuned blocks

### ISSUE THREAD

**Issue by algpy** [SarahRastegar/InfoSieve#1]
Great work on the paper and the overall project!

I went through the repository and have two questions regarding the code. Firstly, I am curious about the evaluation function used in the code; is it 'v1'? Secondly, I would like to know the number of blocks that will be fine-tuned. In the paper, you mentioned fine-tuning the last two blocks, but in the 'contrastive_train.sh' script, it appears that only the last block is fine-tuned.

Thanks in advance.

**Comment by SarahRastegar** [comment]
Thanks a lot!

The evaluation function is 'v2', similar to the GCD paper.
As mentioned in the supplemental, we used the last two blocks for fine-grained datasets for fine-tuning, but for generic datasets in the final version of the paper, we used only the last block. The default values of the contrastive_training.py python file are the ones we've used for the CUB dataset.

**Comment by algpy** [comment]
Thank you for providing these answers; they have been helpful! However, I encountered some new issues when running the code.

Based on your response, does it imply that the "--eval_funcs 'v1' 'v2'" in 'contrastive_train.sh' should be updated to "--eval_funcs 'v2'"? It seems that with "--eval_funcs 'v1' 'v2'", the code only returns the 'v1' indicator. Additionally, when running the code, I only achieved "58.5," "50.1," and "51.5" accuracy on the train unlabeled dataset of CUB, Stanford-Cars, and Aircraft, respectively, with eval_func v2, which differs from the results reported in the paper. However, I achieve "66.8," "55.7," and "56.2" accuracy on the train unlabeled dataset of CUB, Stanford-Cars, and Aircraft, respectively, with eval_func v1. I'm not quite sure what caused this discrepancy. Could you provide the checkpoints of the final model?

Thank you so much!

**Comment by SarahRastegar** [comment]
For training, the code uses 'v1' to choose the best model because we use Kmeans in training to show categories, for evaluation we use 'v2' because SSK accounts for the fact that part of our categories are already known which can provide valid initialization for centers of clusters. 
You should get the paper results on the CUB dataset if you run the "python contrastive_training.py" script without any argument (hence default values) from the command line. After that, the "--dataset_name aircraft" argument can be passed for other datasets, for instance, aircraft.

**Comment by algpy** [comment]
thanks for your reply.
